### PR TITLE
[Enhence](array function) support array_remove function process NULL element

### PR DESCRIPTION
**Pull request by ZhangYu0123**
# Proposed changes
Array remove function can't  processing null element and just return null as result.  In some cases, it isn't logical.  Problems are as follows：

image

So this pr adds functionality for array_remove to handle NULL elements. Details as follows：

1. Added an optional parameter like this:   
    array<T> array_remove(array<T> arr, T val, **is_process_null = false**).  
2. When is_process_null is 1 or True,  array_remove will remove null element in array like this:
 
image

3. When is_process_null is 0 or False or default. array_remove is the same as the previous processing logic.
image

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by ZhangYu0123** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 33.41 seconds
 stream load tsv:          454 seconds loaded 74807831229 Bytes, about 157 MB/s
 stream load json:         38 seconds loaded 2358488459 Bytes, about 59 MB/s
 stream load orc:          73 seconds loaded 1101869774 Bytes, about 14 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230312104359_clickbench_pr_112950.html

**Comment by ZhangYu0123** [comment]
run p0

**Comment by ZhangYu0123** [comment]
run P0

**Comment by ZhangYu0123** [comment]
run p0

**Comment by ZhangYu0123** [comment]
run p0

**Comment by ZhangYu0123** [comment]
run buildall

**Comment by carlvinhust2012** [comment]
So does this pr follow the behavior of Spark or Hive？

File: regression-test/suites/query_p0/sql_functions/array_functions/test_array_functions.groovy [hunk 1]
```diff
@@ -60,6 +60,10 @@ suite("test_array_functions") {
     qt_select "SELECT array_remove(k2, k1), k1 FROM ${tableName} ORDER BY k1"
     qt_select "SELECT array_remove(k8, cast('2023-02-05' as datev2)) FROM ${tableName} ORDER BY k1"
     qt_select "SELECT array_remove(k10, cast('2022-10-15 10:30:00.999' as datetimev2(3))) FROM ${tableName} ORDER BY k1"
+    qt_select "SELECT array_remove(k2, k1, 1), k1 FROM ${tableName} ORDER BY k1"
```
Comment (carlvinhust2012) [inline regression-test/suites/query_p0/sql_functions/array_functions/test_array_functions.groovy:63]: Maybe add some case for decimal and decimalv2 type is more better.

**Comment by cambyzju** [comment]
please also update the docs

**Pull request closed without merging**